client/src/components/Filter.js

```diff
@@ -1,306 +1,293 @@
-		width: FILTER_WIDTH,
-		padding		: theme.spacing(3, 2, 2, 2),
-		/*
-		 * boxShadow: '0px 3px 5px -1px rgba(0,0,0,0.2), 0px 5px 8px 0px rgba(0,0,0,0.14), 0px 1px 14px 0px rgba(0,0,0,0.12)',
-		 * */
-	},
-	close		: {
-		color		: theme.palette.grey[500],
-	},
-	dateInput		: {
-		width		: 158,
-		fontSize		: 14,
-	},
-	button : {
-		marginTop: 5,
-	},
-}}
+  return {
+    root: {},
+    drawer: {
+      flexShrink: 0
+    },
+    drawerPaper: {
+      width: FILTER_WIDTH,
+      padding: theme.spacing(3, 2, 2, 2)
+      /*
+       * boxShadow: '0px 3px 5px -1px rgba(0,0,0,0.2), 0px 5px 8px 0px rgba(0,0,0,0.14), 0px 1px 14px 0px rgba(0,0,0,0.12)',
+       * */
+    },
+    close: {
+      color: theme.palette.grey[500]
+    },
+    dateInput: {
+      width: 158,
+      fontSize: 14
+    },
+    button: {
+      marginTop: 5
+    }
+  };
+};
 
-function Filter(props){
-	console.log(styles);
-	const {classes, filter}		= props
-	//console.error('filter:%o', filter)
-	const dateStartDefault		= '1970-01-01'
-	const dateEndDefault		= `${dateformat(Date.now(), 'yyyy-mm-dd')}`
-	const [treeId, setTreeId]		= useState(filter.treeId)
-	const [planterId, setPlanterId]		= useState(filter.planterId)
-	const [deviceId, setDeviceId]		= useState(filter.deviceId)
-	const [planterIdentifier, setPlanterIdentifier]		= useState(filter.planterIdentifier)
-	const [status, setStatus]		= useState(filter.status)
-	const [approved, setApproved]		= useState(filter.approved)
-	const [active, setActive]		= useState(filter.active)
-	const [dateStart, setDateStart]		= useState(filter.dateStart || dateStartDefault)
-	const [dateEnd, setDateEnd]		= useState(filter.dateEnd || dateEndDefault)
-	//console.error('the tree id:%d', treeId)
+function Filter(props) {
+  const { classes, filter } = props;
+  const dateStartDefault = '1970-01-01';
+  const dateEndDefault = `${dateformat(Date.now(), 'yyyy-mm-dd')}`;
+  const [treeId, setTreeId] = useState(filter.treeId);
+  const [planterId, setPlanterId] = useState(filter.planterId);
+  const [deviceId, setDeviceId] = useState(filter.deviceId);
+  const [planterIdentifier, setPlanterIdentifier] = useState(
+    filter.planterIdentifier
+  );
+  const [status, setStatus] = useState(filter.status);
+  const [approved, setApproved] = useState(filter.approved);
+  const [active, setActive] = useState(filter.active);
+  const [dateStart, setDateStart] = useState(
+    filter.dateStart || dateStartDefault
+  );
+  const [dateEnd, setDateEnd] = useState(filter.dateEnd || dateEndDefault);
 
-	function handleDateStartChange(e){
-		setDateStart(e.target.value || dateStartDefault)
-	}
+  const handleDateStartChange = date => {
+    const formattedDate = dateformat(date, 'yyyy-mm-dd');
+    setDateStart(formattedDate);
+  };
 
-	function handleDateEndChange(e){
-		setDateEnd(e.target.value || dateEndDefault)
-	}
+  const handleDateEndChange = date => {
+    const formattedDate = dateformat(date, 'yyyy-mm-dd');
+    setDateEnd(formattedDate);
+  };
```

Comment:
Changed `handleDateStartChange` and `handleDateEndChange` for the date picker change.